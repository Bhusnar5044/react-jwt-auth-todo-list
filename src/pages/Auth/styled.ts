import { Box } from '@components/Lib';
import { theme } from '@theme';
import { styled } from '@utils';

export const StyledBox = styled(Box)`
    width: 80%;
    margin: auto;
    padding: 5rem;
    border-radius: 2rem;
    background-color: ${theme.colors.primary.main};
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    @media only screen and (min-width: 1200px) {
        width: 40%;
    }
`;

export const StyledForm = styled.form`
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 2rem;
    width: 100%;
`;
