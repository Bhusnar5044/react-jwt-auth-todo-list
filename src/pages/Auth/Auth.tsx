import { Box, Button, Text, Textfield } from '@Lib';
import { PageLayout } from '@components/Layout';
import { AppBar } from '@components/Layout/AppBar/AppBar';
import { urls } from '@constant/urls';
import { useAuth } from '@context/AuthProvider';
import { theme } from '@theme';
import { emailValidator, fetch } from '@utils';
import { ChangeEvent, FC, FormEvent, memo, useCallback, useEffect, useState } from 'react';
import { StyledBox, StyledForm } from './styled';

export const Auth: FC = memo(() => {
    const [username, setUsername] = useState('');
    const [pass, setPass] = useState('');
    const [isDisabled, setIsDisabled] = useState(true);
    const [isLoading, setIsLoading] = useState(false);
    const [error, setError] = useState('');
    const { login } = useAuth();
    const onChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
        const { name, value } = event.target;
        if (name === 'username') setUsername(value);
        else setPass(value);
    }, []);
    const handleSubmit = useCallback(
        async (event: FormEvent) => {
            event.preventDefault();
            // used mock API fixed email pass
            const config = {
                url: urls.login,
                data: {
                    email: username,
                    password: pass,
                },
                method: 'POST',
            };

            setIsLoading(true);

            const { response, error } = await fetch(config, '', true);

            if (response) return login?.({ tokens: response.data });
            if (error) {
                console.log(error.data.message);
                setError('Please enter valid email and password');
            }

            setIsLoading(false);
        },
        [login, pass, username],
    );

    useEffect(() => {
        if (username && pass) setIsDisabled(false);
    }, [pass, username]);

    return (
        <PageLayout>
            <AppBar />
            <Box justifyContent="center" alignItems="center">
                <StyledBox flexDirection="column" gap="3rem" justifyContent="center" alignItems="center">
                    <Text textColor="#fff" textVariant="h4" textWeight="Strong">
                        Login
                    </Text>
                    <StyledForm onSubmit={handleSubmit}>
                        <Textfield
                            fullWidth
                            type="text"
                            label="Username"
                            name="username"
                            value={username}
                            onChange={onChange}
                            required
                            validator={emailValidator}
                        />
                        <Textfield
                            fullWidth
                            type="password"
                            label="Password"
                            name="pass"
                            value={pass}
                            onChange={onChange}
                            required
                        />
                        <Button type="submit" disabled={isDisabled} loading={isLoading}>
                            Submit
                        </Button>
                        <Text textColor={theme.colors.error.light}>{error}</Text>
                    </StyledForm>
                </StyledBox>
            </Box>
        </PageLayout>
    );
});

Auth.displayName = 'Auth';
