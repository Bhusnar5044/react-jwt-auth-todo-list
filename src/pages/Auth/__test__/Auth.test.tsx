import { render } from '@test-utils';
import { Auth } from '../Auth';

describe('Auth Page', () => {
    it('should render properly', () => {
        const { container } = render(<Auth />);
        expect(container).toMatchSnapshot();
    });
});
