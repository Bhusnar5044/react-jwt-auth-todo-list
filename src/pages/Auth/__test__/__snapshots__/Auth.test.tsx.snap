// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page should render properly 1`] = `
.c3 {
  display: flex;
  align-items: center;
  gap: 2rem;
}

.c6 {
  display: flex;
  justify-content: center;
  align-items: center;
}

.c7 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 3rem;
}

.c11 {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
}

.c18 {
  position: relative;
  display: inline-block;
  outline: none;
  cursor: not-allowed;
  text-align: center;
  border: unset;
  padding: 1rem 2rem;
  color: #fff;
  background: #008085;
  border-radius: 0.8rem;
  font-weight: 600;
  font-size: 1.4rem;
  line-height: 2.2rem;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 0px 0px,rgba(0, 0, 0, 0.2) 0px -1px 0px 0px inset;
  opacity: 0.7;
  pointer-events: none;
}

.c18:hover {
  background: #4caf50;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2),0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.c18.loading {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.7;
}

.c18.loading::after {
  content: '';
  position: absolute;
  width: 2rem;
  height: 2rem;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  border: 0.4rem solid transparent;
  border-top-color: #1565c0;
  border-right-color: #1565c0;
  border-radius: 50%;
  animation: gNWkkb 1s ease infinite;
}

.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c9 {
  margin: 0;
  color: #fff;
  font-size: 1.8rem;
  font-weight: 700;
  letter-spacing: -0.02rem;
  line-height: 2.4rem;
  text-align: initial;
}

.c19 {
  margin: 0;
  color: #ef5350;
  font-size: 1.4rem;
  font-weight: 400;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c17 {
  position: absolute;
  top: 20px;
  left: 12px;
  font-size: 16px;
  color: rgba(#212529, 0.5);
  font-weight: 500;
  transform-origin: 0 0;
  transform: translate3d(0, 0, 0);
  transition: all 0.2s ease;
  pointer-events: none;
}

.c15 {
  -webkit-appearance: none;
  appearance: none;
  width: 100%;
  border: 0;
  font-family: inherit;
  padding: 16px 12px 0 12px;
  height: 56px;
  font-size: 16px;
  font-weight: 400;
  background: rgba(#212529, 0.02);
  box-shadow: inset 0 -1px 0 rgba(#212529, 0.3);
  color: #212529;
  transition: all 0.15s ease;
}

.c15:hover {
  background: rgba(#212529, 0.04);
  box-shadow: inset 0 -1px 0 rgba(#212529, 0.5);
}

.c15:not(:placeholder-shown)+.c16 {
  color: rgba(#212529, 0.5);
  transform: translate3d(0, -12px, 0) scale(0.75);
}

.c15:focus {
  background: rgba(#212529, 0.05);
  outline: none;
  box-shadow: inset 0 -2px 0 #1976d2;
}

.c15:focus +.c16 {
  color: #1976d2;
  transform: translate3d(0, -12px, 0) scale(0.75);
}

.c15:focus +.c16 +.focus-bg {
  transform: scaleX(1);
  transition: all 0.1s ease;
}

.c13 {
  position: relative;
  border-radius: 3px;
  overflow: hidden;
  width: 100%;
}

.c13 .focus-bg {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(#6c757d, 0.05);
  z-index: -1;
  transform: scaleX(0);
  transform-origin: left;
}

.c13 .c14 {
  width: 100%;
}

.c12 {
  width: 100%;
}

.c1 {
  display: flex;
  align-items: center;
  width: 100%;
  margin: auto;
  background-color: #008085;
  padding: 1rem 0;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 90%;
  margin: auto;
  color: #fff;
}

.c2 a:hover {
  color: #495057;
}

.c4 {
  width: 2.5rem;
  height: auto;
}

.c0 {
  display: grid;
  height: 100vh;
  width: 100vw;
  grid-template-areas: 'header' 'main';
  grid-template-columns: 1fr;
  grid-template-rows: 4.2em auto;
  background-color: #eff2f4;
}

.c0>div {
  padding: 0;
  margin: 2rem 0;
}

.c0>header {
  grid-area: header;
}

.c0>main {
  grid-area: main;
  transition: width 200ms;
  padding-bottom: 35px;
}

.c8 {
  width: 80%;
  margin: auto;
  padding: 5rem;
  border-radius: 2rem;
  background-color: #1976d2;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2),0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.c10 {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
  width: 100%;
}

@media only screen and (min-width: 1200px) {
  .c8 {
    width: 40%;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          alignitems="center"
          class="c3"
          gap="2rem"
        >
          <img
            class="c4"
            src="/vite.svg"
          />
          <span
            class="c5"
            textalign="initial"
            textcolor="inherit"
          >
            React JWT Demo
          </span>
        </div>
      </div>
    </header>
    <div
      alignitems="center"
      class="c6"
      justifycontent="center"
    >
      <div
        alignitems="center"
        class="c7 c8"
        flexdirection="column"
        gap="3rem"
        justifycontent="center"
      >
        <h4
          class="c9"
          textalign="initial"
          textcolor="#fff"
          textvariant="h4"
          textweight="Strong"
        >
          Login
        </h4>
        <form
          class="c10"
        >
          <div
            class="c11 c12"
            flexdirection="column"
            gap="0.5rem"
          >
            <label
              class="c13"
              for="inp"
            >
              <input
                class="c14 c15"
                id="inp"
                name="username"
                placeholder=" "
                required=""
                type="text"
                value=""
              />
              <label
                class="c16 c17"
              >
                Username
              </label>
              <span
                class="focus-bg"
              />
            </label>
          </div>
          <div
            class="c11 c12"
            flexdirection="column"
            gap="0.5rem"
          >
            <label
              class="c13"
              for="inp"
            >
              <input
                class="c14 c15"
                id="inp"
                name="pass"
                placeholder=" "
                required=""
                type="password"
                value=""
              />
              <label
                class="c16 c17"
              >
                Password
              </label>
              <span
                class="focus-bg"
              />
            </label>
          </div>
          <button
            class="c18"
            disabled=""
            type="submit"
          >
            Submit
          </button>
          <span
            class="c19"
            textalign="initial"
            textcolor="#ef5350"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
